### PR TITLE
First Blog

### PR DESCRIPTION
**Pull request by shilpiprd**
This folder should provide a place to put all the blog posts.

### CODE REVIEW

**Comment by shilpiprd** [comment]
@skoudoro. I'm actually having trouble writing up a rst file. Since DIPY is very correlated to FURY, i figured that the structure of the rst file would be similar, but it turns out it's not. 
I tried running my .rst file with the sphinx version we have in dipy, but I got error in docscrape.py file, as an import was outdated for all python version > 3.10. 
I think some of the files for sphinx are a little outdated and needs to be probably modified. 
Also, can you help me with the starting structure of the rst files  ?

**Comment by skoudoro** [comment]
Hi @shilpiprd,

Thank you for this first blog post! I will do my first review after @lb-97 and @RafaelNH.

**Comment by skoudoro** [comment]
> i figured that the structure of the rst file would be similar, but it turns out it's not.

it is correlated and same syntax. it is just missing an extension (`ablog`) that we need to add in this repository.

**Comment by shilpiprd** [comment]
> 

Yeah I actually figured that out, but I wasn't sure if it was ok for me to change an already existing file.
 Also, the file : doc/sphinxext/docscrape.py   Line#88 in this file imports collections.Mappinngs, but for alll python versions > 3.10, it should be collections.abc.Mappings. 
This actually gave me error on my system. 
Should I change that as well ?

**Comment by skoudoro** [comment]
> Should I change that as well ?

You add a check if python >= 3.10  import one version otherwise, the other

**Comment by shilpiprd** [comment]
Hey Serge, I think this PR's commit history doens't look like it should, so should i create a new PR which contains all the required changes ? It would look cleaner that way.

**Comment by skoudoro** [comment]
Hi @shilpiprd,

Use this PR as a training one. 

You can squash the commit to reduce the number of commit and improve commit history. 
Please try to do that here.

Do NOT create a new PR.  I assure you that all you future PR will be easier to manage after learning this

**Comment by shilpiprd** [comment]
I think I squahsed all the needed commits together, "2nd_Final_commit " should be good.

**Comment by lb-97** [comment]
@shilpiprd, I think you can squash all commits to a single commit, that should suffice.

**Comment by shilpiprd** [comment]
Are you sure ? Because the structure of the rst file has been removed in 2nd commit, but it's there in the final squashed commit. If I squash them then wouldn't 2nd commit overwrite all changes (i.e remove the top structural part )? Same for 1st commit, I've changed some lines in the rst blog post.

**Comment by lb-97** [comment]
If you squash them using `git rebase -i HARD~3` then there shouldn't be any overriding. All commits are squashed to one change, yet to be staged with a new commit message. @shilpiprd

**Comment by shilpiprd** [comment]
yeah, I think it's done now. Thanks @lb-97 !

**Review by lb-97 — APPROVED** [review]

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
+My Journey to GSoC 2022
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:1]: 2022 ???

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+
+About Myself
+~~~~~~~~~~~~
+
+Hey there! I'm Shilpi, a Computer Science and Engineering undergrad at Dayananda Sagar College of Engineering, Bangalore. I'm on track to grab my degree in 2024.
+My relationship with Python started just before I started college - got my hands dirty with this awesome Python Specialization course on Coursera. 
+When it comes to what makes me tick, it's all things tech. I mean, new technology always excites me. Ubuntu, with its fancy terminal and all, used to intimidate me at first, but now, I get a thrill out of using it to do even the simplest things.
+Up until 2nd year I used to do competitive programming and a bit of ML.But from 3rd year I've been into ML very seriously, doing several courses on ML as well solving ML problems on kaggle. ML is very fun and I've done a few project on ML as well.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:17]: `ML.But`  missing space

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+
+About Myself
+~~~~~~~~~~~~
+
+Hey there! I'm Shilpi, a Computer Science and Engineering undergrad at Dayananda Sagar College of Engineering, Bangalore. I'm on track to grab my degree in 2024.
+My relationship with Python started just before I started college - got my hands dirty with this awesome Python Specialization course on Coursera. 
+When it comes to what makes me tick, it's all things tech. I mean, new technology always excites me. Ubuntu, with its fancy terminal and all, used to intimidate me at first, but now, I get a thrill out of using it to do even the simplest things.
+Up until 2nd year I used to do competitive programming and a bit of ML.But from 3rd year I've been into ML very seriously, doing several courses on ML as well solving ML problems on kaggle. ML is very fun and I've done a few project on ML as well. 
+Coding? Absolutely love it. It's like, this is what I was meant to do, y'know? I got introduced to git and GitHub in my first year - was super curious about how the whole version control thing worked. And then, I stumbled upon the world of open source in my second year and made my first contribution to Tardis.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:18]: tardis: add a link

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+So, I started off finding out about GSoC - how many hours do selected folks put in, the kind of projects people usually tackle, and all that stuff. To get a handle on what they want in a proposal, I turned to some successful ones from previous years. Really gave me an idea of the kind of stuff they expect you to bring to the table.
+Trying to find the organization that'd go with my skill set, I stumbled upon Python Software Foundation, and I was like, "This is it!". And under PSF, there was DIPY. 
+Diving into DIPY's docs was a breeze as they've got it so well put together that I managed to get my head around a completely new topic, "Diffusion MRI", just by going through their introductory docs and a bit of gpt.
+While exploring DIPY, I noticed this issue that needed a new feature. It took a good bit of reading to really understand what they were looking for and how to actually build that feature. And then, I submitted my first PR (`check it out here: https://github.com/dipy/dipy/pull/2749`)! Getting it merged wasn't exactly easy - there was a lot of room for improvement up in my code, but honestly,I feel like it's all part of the learning curve.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:28]: - check it out here:   link is not working. or it is not intended to be a link ?
- `honestly,I `. missing space

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+I was a bit of a latecomer to GSoC, so I didn't have much time to make up a ton of PRs. Plus, by the time I'd submitted my first PR, the proposal submission period had already begun. So, I focused all my energy on increasing my knowledge on the topic and polishing my proposal.Plus, I'd wanted to get my proposal reviewed at least once before I submitted it.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:29]: `proposal.Plus,` missing space

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+Code contributions:
+
+1. [https://github.com/dipy/dipy/pull/2749]
+
+
+The Day
+~~~~~~~
+
+May 4th: I woke up feeling like a nervous wreck. That interview with my organization? Let's just say it didn't go very well. Yet, I couldn't help but hope for the best. The results were supposed to drop at 11:45pm, a moment I wasn't exactly looking forward to.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:39]: > Let's just say it didn't go very well

I confirm.... lol

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+I tried logging into Google to check, but couldn't. Too many people doing the same thing. I threw my hands up, gave up on the login battle, and got back to work, hoping to distract myself.
+Fast forward to 1:30am - I figured by now, the log-in rush should have calmed down. I gave it another shot and... I got in! I clicked over to the dashboard, and there it was. My project. Right there, listed under the Projects section.I had heard that if you get selected, your proposal shows up there.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:41]: `section.I` missing space

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+To confirm that it was actually happening, I picked my phone to check if I'd gotten any official email yet.And yes!! I'd gotten it at 12:49 am. I just hadn't checked.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:42]: `yet.And` missing space

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+I whooped, woke up my roomies, rushed to call my parents.
+Honestly, words can't even begin to capture how I felt at that moment.
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:44]: `me.It` missing space

File: doc/sphinxext/docscrape.py [hunk 10]
```diff
@@ -8,6 +8,11 @@
 import collections
 import sys
 
+if sys.version_info >= (3, 10):
```
Comment (skoudoro) [inline doc/sphinxext/docscrape.py:11]: This foes not always work well, specially with dev version  (like `3.10.1dev0`).

Ca you use packaging like we do [here](https://github.com/dipy/dipy/blob/89e4ab6a4a2c40f8681e02f930a3fc2371f51f82/dipy/conftest.py#L15)

**Review by skoudoro — COMMENTED** [review]
Hi @shilpiprd,

See below my review. After that, it should be ready to go.

@lb-97, can you look at my review also to get inspired for the future one. 

Thanks to both of you

File: doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst (hunk 6 above)
Comment (shilpiprd) [reply doc/posts/2023/2023_05_19_Shilpi_week0_journey_to_GSoC.rst:39]: so you do agree, huh

File: doc/sphinxext/docscrape.py (hunk 10 above)
Comment (shilpiprd) [reply doc/sphinxext/docscrape.py:11]: This foes not always work well, specially with dev version (like 3.10.1dev0).
What do you mean by foes ?

Comment (skoudoro) [reply doc/sphinxext/docscrape.py:11]: sorry, typo. 

it is `does` and not `foes`

Comment (shilpiprd) [reply doc/sphinxext/docscrape.py:11]: Is this fine ? 
```
from packaging.version import Version
if Version(sys.__version__) >= Version('3.10'):
    from collections.abc import Mapping 
else:
    from collections import Mapping 
```

Comment (skoudoro) [reply doc/sphinxext/docscrape.py:11]: yes

**Review by skoudoro — COMMENTED** [review]
be careful, you committed [doc/posts/2023/2023_05_29_Shilpi_week1.pdf](https://github.com/dipy/dipy/pull/2813/files#diff-40834b5d4e63950f2b9af40141268b0fcd5ce83c80a74f6b94a9bef343ebb4cc) accidentally.
to remove

**Comment by shilpiprd** [comment]
> Member

yeah it's supposed to be an rst file right?

**Comment by lb-97** [comment]
> @lb-97, can you look at my review also to get inspired for the future one.

Sure @skoudoro. Thankyou.

File: doc/posts/2023/2023_05_29_Shipi_week1.rst [hunk 11]
```diff
@@ -0,0 +1,48 @@
+Community Bonding & Week0:
+=======================
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_29_Shipi_week1.rst:2]: not enough long, it should be the same size as the text above

File: doc/posts/2023/2023_05_29_Shipi_week1.rst [hunk 12]
```diff
@@ -0,0 +1,48 @@
+Community Bonding & Week0:
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_29_Shipi_week1.rst:1]: - colon to remove
- Title to improve `& week0` ???

File: doc/posts/2023/2023_05_29_Shipi_week1.rst [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+=======================
+
+.. post:: May 19 2023
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_29_Shipi_week1.rst:4]: wrong date

File: doc/posts/2023/2023_05_29_Shipi_week1.rst [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    :author: Shilpi Prasad
+    :tags: google
+    :category: gsoc
+
+
+
+What I did this week:
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_29_Shipi_week1.rst:11]: colon to remove

File: doc/posts/2023/2023_05_29_Shipi_week1.rst [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+~~~~~~~~~~~~~~~~~~~
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_29_Shipi_week1.rst:12]: not enough long, it should be the same size as the text above

File: doc/posts/2023/2023_05_29_Shipi_week1.rst [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+I also submitted this PR: `https://github.com/dipy/dipy/pull/2813` . In this PR members of my organization are supposed to submit all the
+blogs.
+But mostly I spent a lot of time in implementing the already existing MultiTensor Simulation on my local system , and also completing the assignment which my mentor gave me.
+In this assignment, I was given a specific number of directions, 'n' and some steps on how to produce bvals and bvecs. I had to create gtab1 and gtab2. and then taking gtab1 & gtab2 as input, I was supposed to create a function which would give output btensor i.e
+btens.
+The purpose of this assignment was to strengthen my knowledge on concepts I've already read
+and also to give me some coding experience, as this is critical in order for me to be able to
+implement the rest of my project.
+
+What is coming up next Week:
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_29_Shipi_week1.rst:28]: colon to remove

File: doc/posts/2023/2023_05_29_Shipi_week1.rst [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+
+These simulations were basically the first task of the proposal.
+So after the btensor I intend on producing the synthetic signals using the qti model (hint on how
+it is done in qti tests).
+make a figure similar to figure 1 of the 2021 CTI paper:
+https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.28938
+
+
+Did I get stuck anywhere
+~~~~~~~~~~~~
```
Comment (skoudoro) [inline doc/posts/2023/2023_05_29_Shipi_week1.rst:39]: not enough long, it should be the same size as the text above

**Review by skoudoro — COMMENTED** [review]
Hi @shilpiprd,

See below my review. Also, link does not work. Please check how to do a link in rst. 

Thanks

**Comment by shilpiprd** [comment]
I hope with this all the links would work, please let me know if any other  changes are required.

**Comment by skoudoro** [comment]
Thanks @shilpiprd , merging

**Pull request merged**